common/testing/taskpoller/taskpoller.go

```diff
@@ -125,9 +146,123 @@ func (p *TaskPoller) PollAndHandleWorkflowTask(
 // If no task is available, it returns NoTaskAvailable.
 func (p *workflowTaskPoller) HandleTask(
```

Comment:
Most of the time we do not work with queries, it seems a little to much to impose the nested TaskCompletedResponse on all usages. Why not keep all these methods untouched and add a `HandleQueryTask` instead:
```
func (p *workflowTaskPoller) HandleQueryTask(tv *testvars.TestVars,
	handler func(task *workflowservice.PollWorkflowTaskQueueResponse) (*workflowservice.RespondQueryTaskCompletedResponse, error)
```